dataset/test/bin_test.cpp

```diff
@@ -366,3 +366,26 @@ TEST_P(BinTest, rebin_various_edges_1d) {
     for (const auto &e1 : edges)
       bin(bin(table, {e0}), {e1});
 }
+
+TEST_P(BinTest, clear_binning_and_bin_along_different_dimension) {
```

Comment:
What happens if we clear all existing binnings but don't add a new one? Do we get a scalar binned variable as a result?

Reply:
An error is thrown if you don't provide either edges or groups to `bin()`.